finetune/base.py

```diff
@@ -360,8 +360,23 @@ def cached_predict(self):
         self._cached_predict = False
         self.close()
 
+    def _sort_by_length(self, Xs):
+        """
+        Returns the sorted array and the idxs to invert the sort operation
+        """
+        lengths = [len(X) for X in Xs]
+        sorted_idxs = np.argsort(lengths)
+        invert_idxs = np.zeros(sorted_idxs.shape, dtype=int)
+        invert_idxs[sorted_idxs] = np.arange(sorted_idxs.shape[0])
+        return np.asarray(Xs)[sorted_idxs], invert_idxs
+
     def _inference(self, Xs, predict_keys=None, n_examples=None, context=None, update_hook=None):
+
         Xs = self.input_pipeline._format_for_inference(Xs)
+        
+        if self.config.sort_by_length:
```

Comment:
So because this sort happens prior to chunk long sequences being applied it will help only in the case that some items are short enough not to get chunked but will not help to group the short chunks?

Reply:
Yup aware of this.  But I actually don't mind that much, because as soon as everything is long enough to get chunked every time this performance hit matters less.  Mostly looking to make the scenario where most everything is short sans a few items faster.